packages/connect-conformance/src/conformance.ts

```diff
@@ -29,34 +29,25 @@ import { execFileSync } from "node:child_process";
 import { fetch } from "undici";
 import { scripts } from "../package.json";
 
-// Extract conformance runner version from the `generate` script
-const [, version] = /conformance:(v\d+\.\d+\.\d+)/.exec(scripts.generate) ?? [
-  "?",
-];
-
-const downloadUrl = `https://github.com/connectrpc/conformance/releases/download/${version}`;
-
 export async function run() {
-  const { archive, bin } = getArtifactNameForEnv();
-  const tempDir = getTempDir();
+  // Extract conformance runner version from the `generate` script
+  const [, version] = /conformance:(v\d+\.\d+\.\d+)/.exec(scripts.generate) ?? [
+    "?",
+  ];
+  const { archive, bin } = getArtifactNameForEnv(version);
+  const tempDir = getTempDir(version);
   const binPath = joinPath(tempDir, bin);
-  if (!existsAndMatchesVersion(binPath)) {
+  if (!existsSync(binPath)) {
+    const downloadUrl = `https://github.com/connectrpc/conformance/releases/download/${version}/${archive}`;
     const archivePath = joinPath(tempDir, archive);
-    await download(`${downloadUrl}/${archive}`, archivePath);
+    await download(downloadUrl, archivePath);
     await extractBin(archivePath, binPath);
   }
   execFileSync(binPath, process.argv.slice(2), {
     stdio: "inherit",
   });
 }
 
-function existsAndMatchesVersion(bin: string) {
-  return (
-    existsSync(bin) &&
-    execFileSync(bin, ["--version"]).toString().endsWith(version)
```

Comment:
The exec output ends with a newline, but `version` does not, so this condition is never true.

Could be fixed with `toString().trim()`, but I chose to remove the `exec` call by adding the version to the bin path instead.